packages/babel-preset-gatsby/src/index.js

```diff
@@ -62,19 +64,44 @@ module.exports = function preset(_, options = {}) {
       ],
     ],
     plugins: [
+      [
+        resolve(`@babel/plugin-transform-destructuring`),
+        {
+          loose: false,
+          selectiveLoose: [
+            `useState`,
+            `useEffect`,
+            `useContext`,
+            `useReducer`,
+            `useCallback`,
+            `useMemo`,
+            `useRef`,
+            `useImperativeHandle`,
+            `useLayoutEffect`,
+            `useDebugValue`,
+          ],
+        },
+      ],
       [
         resolve(`@babel/plugin-proposal-class-properties`),
         {
           loose: true,
         },
       ],
+      [
+        resolve(`@babel/plugin-proposal-object-rest-spread`),
+        {
+          useBuiltIns: true,
+        },
+      ],
       resolve(`babel-plugin-macros`),
       resolve(`@babel/plugin-syntax-dynamic-import`),
       [
         resolve(`@babel/plugin-transform-runtime`),
         {
           helpers: true,
           regenerator: true,
+          useESModules,
```

Comment:
I don't understand this change — what's the reasoning? It doesn't seem to be something CRA is doing.

Reply:
CRA actually does include this [here](https://github.com/facebook/create-react-app/blob/265c1592dcf3853122dcdd50e1bfc1e43f697fbe/packages/babel-preset-react-app/create.js#L32-L36).

And [here](https://babeljs.io/docs/en/babel-plugin-transform-runtime#useesmodules) is some more information about it. So basically since we use webpack (which can handle es modules) we can turn on `useESModules` to save some bytes.